### PR TITLE
[Changelog] NFC: Add an entry about SE-0299

### PR DESCRIPTION
**Pull request by xedin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,26 @@ CHANGELOG
 Swift Next
 ----------
 
+* [SE-0299][]:
+
+  It's now possible to use leading-dot syntax in generic contexts if parameter has a protocol requirement and static members are declared in protocol extension where `Self` is bound a concrete type:
```
Comment (slavapestov) [inline CHANGELOG.md:33]: How about

> It is now possible to use leading-dot syntax in generic contexts to access static members of protocol extensions where `Self` is constrained to a concrete type:

Comment (xedin) [reply CHANGELOG.md:33]: Sounds good!

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,26 @@ CHANGELOG
...
+
+  ```swift
+  public protocol ToggleStyle { ... }
+
+  public struct DefaultToggleStyle: ToggleStyle { ... }
+
+  exntesion ToggleStyle where Self == DefaultToggleStyle {
```
[the remark this answers is not in the document]
Comment (xedin) [reply CHANGELOG.md]: ✅

File: CHANGELOG.md (hunk 1 above)
Comment (xedin) [reply CHANGELOG.md]: Fixed!

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test Windows platform

File: CHANGELOG.md [hunk 3]
```diff
@@ -28,6 +28,26 @@ CHANGELOG
...
+  public protocol ToggleStyle { ... }
+
+  public struct DefaultToggleStyle: ToggleStyle { ... }
+
+  extension ToggleStyle where Self == DefaultToggleStyle {
+    public static var `default`: Self { .init() }
+  }
+
+  struct Toggle {
+    func applyToggle<T: ToggleStyle(_ style: T) { ... }
```
Comment (AnthonyLatsis) [inline CHANGELOG.md:45]: Missing '>'

File: CHANGELOG.md [hunk 4]
```diff
@@ -28,6 +28,26 @@ CHANGELOG
...
+* [SE-0299][]:
+
+  It is now possible to use leading-dot syntax in generic contexts to access static members of protocol extensions where `Self` is constrained to a concrete type:
+
+  ```swift
+  public protocol ToggleStyle { ... }
+
+  public struct DefaultToggleStyle: ToggleStyle { ... }
+
+  extension ToggleStyle where Self == DefaultToggleStyle {
```
Comment (AnthonyLatsis) [inline CHANGELOG.md:40]: Perhaps we should mention that the type must be fully concrete (I suppose `Self == Foo<Assoc>` will not work).

File: CHANGELOG.md (hunk 3 above)
Comment (xedin) [reply CHANGELOG.md:45]: Ugh, thank you! So much for typing everything out...

Comment (xedin) [reply CHANGELOG.md]: ✅

File: CHANGELOG.md (hunk 4 above)
Comment (xedin) [reply CHANGELOG.md:40]: ✅

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test macOS platform

**Comment by xedin** [comment]
@swift-ci please smoke test Windows platform

**Pull request merged**